Sources/StreamChat/WebSocketClient/Events/Event.swift

```diff
@@ -14,39 +14,107 @@ protocol EventWithPayload: Event {
     var payload: Any { get }
 }
 
-/// A protocol for user `Event` where it has a user payload.
-protocol EventWithUserPayload: EventWithPayload {
+/// A protocol for any `UserEvent` where it has a `user` payload.
+protocol UserEvent: EventWithPayload {
     var userId: UserId { get }
 }
 
-/// A protocol for a current user `Event` where it has `me` payload.
-protocol EventWithCurrentUserPayload: EventWithPayload {
+/// A protocol for any `UserEvent` where it has a `user` payload and is associated with Channel.
+protocol UserChannelSpecificEvent: UserEvent {
+    var cid: ChannelId { get }
+}
+
+/// A protocol for any `ChannelEvent` where it has a  `channel` payload.
+protocol ChannelEvent: EventWithPayload {
+    var cid: ChannelId { get }
+}
+
+/// A protocol for any `MemberEvent` where it has a `member`, and `channel` payload.
+public protocol MemberEvent: Event {
+    var memberUserId: UserId { get }
+    var cid: ChannelId { get }
+}
+
+/// A protocol for any `MessageEvent` where it has a `user`, `channel` and `message` payloads.
+protocol MessageEvent: EventWithPayload {
+    var userId: UserId { get }
+    var cid: ChannelId { get }
+    var messageId: MessageId { get }
+}
+
+/// A protocol for any  `ReactionEvent` where it has reaction with `message`, `channel`, `user` and `reaction` payload.
+protocol ReactionEvent: EventWithPayload {
+    var userId: UserId { get }
+    var cid: ChannelId { get }
+    var reactionType: MessageReactionType { get }
+    var reactionScore: Int { get }
+}
+
+/// A protocol for `TypingEvent` which contains `user`, `channel` payloads. Also differs whether it's start or stop by `isTyping` property.
+protocol UserTypingEvent: EventWithPayload {
+    var userId: UserId { get }
+    var cid: ChannelId { get }
+    var isTyping: Bool { get }
+}
+
+protocol UserNotificationEvent: EventWithPayload {
+    var userId: UserId { get }
+}
```

Comment:
I don't think we need to distinguish between normal and notification events. This applies for all other "notification" protocols